### PR TITLE
crush: update documentation for negative choose step

### PR DESCRIPTION
**Pull request by dachary**
Signed-off-by: Loic Dachary <loic@dachary.org>

### CODE REVIEW

File: src/crush/builder.h [hunk 1]
```diff
@@ -87,10 +87,10 @@ extern struct crush_rule *crush_make_rule(int len, int ruleset, int type, int mi
  *     __arg1__ leaves within all the buckets of type __arg2__ and
  *     select them.
  *
- * In all __CHOOSE__ steps, if __arg1__ is zero, the number of items
- * to select is determined by the __max_result__ argument of
- * crush_do_rule(), i.e. __arg1__ is __max_result__ minus the number of
- * items already in the result.
+ * In all __CHOOSE__ steps, if __arg1__ is lower or equal to zero, the
```
Comment (smithfarm) [inline src/crush/builder.h]: s/lower or equal to/less than or equal to/

File: src/crush/builder.h [hunk 2]
```diff
@@ -87,10 +87,10 @@ extern struct crush_rule *crush_make_rule(int len, int ruleset, int type, int mi
...
+ * number of items to select is the __max_result__ argument of
+ * crush_do_rule() minus __arg1__. It is common to set __arg1__ to
```
Comment (smithfarm) [inline src/crush/builder.h]: Either:

"The number of items to select is determined by subtracting __arch1__ from the __max_result__ argument of crush_do_rule()."

or

"The number of items to select is equal to the __max_result__ argument of crush_do_rule() minus __arg1__."

Comment (smithfarm) [inline src/crush/builder.h]: @dachary Maybe the third version (just added) is best.

Comment (dachary) [inline src/crush/builder.h]: The third version is incorrect: only __max_result__ is an argument of do_rule. The __arg1__ is an argument of the rulestep. I like the second version actually, much clearer !

**Comment by dachary** [comment]
@smithfarm updated and repushed !

**Pull request merged**